src/rpc/client.cpp

```diff
@@ -186,39 +186,62 @@ static const CRPCConvertParam vRPCConvertParams[] =
     { "addpeeraddress", 1, "port"},
     { "stop", 0, "wait" },
 };
+/**
+ * Specify a (method, idx, name) here if the RPC argument is a blockhash
+ * and can meaningfully be abbreviated as a height (eg @1000)
+ *
+ * @note Parameter indexes start from 0.
+ */
+static const CRPCConvertParam rpc_params_convert_blockhash[] =
+{
+    { "getblockheader", 0, "blockhash" },
+    { "getblock", 0, "blockhash" },
+    { "preciousblock", 0, "blockhash" },
+    { "invalidateblock", 0, "blockhash" },
+    { "getchaintxstats", 1, "blockhash" },
+    { "getblockfilter", 0, "blockhash" },
+    { "getrawtransaction", 2, "blockhash" },
+    { "gettxoutproof", 1, "blockhash" },
+};
 // clang-format on
 
+namespace {
 class CRPCConvertTable
 {
 private:
     std::set<std::pair<std::string, int>> members;
     std::set<std::pair<std::string, std::string>> membersByName;
 
+    void setup(size_t n_elem, const CRPCConvertParam *params);
+
 public:
-    CRPCConvertTable();
 
-    bool convert(const std::string& method, int idx) {
+    template<int N>
+    inline CRPCConvertTable(const CRPCConvertParam (&params)[N]) {
+        setup(N, params);
+    }
+
+    bool convert(const std::string& method, int idx) const {
         return (members.count(std::make_pair(method, idx)) > 0);
     }
-    bool convert(const std::string& method, const std::string& name) {
+    bool convert(const std::string& method, const std::string& name) const {
         return (membersByName.count(std::make_pair(method, name)) > 0);
     }
 };
+};
 
-CRPCConvertTable::CRPCConvertTable()
+void CRPCConvertTable::setup(const size_t n_elem, const CRPCConvertParam* params)
 {
-    const unsigned int n_elem =
-        (sizeof(vRPCConvertParams) / sizeof(vRPCConvertParams[0]));
-
-    for (unsigned int i = 0; i < n_elem; i++) {
-        members.insert(std::make_pair(vRPCConvertParams[i].methodName,
-                                      vRPCConvertParams[i].paramIdx));
-        membersByName.insert(std::make_pair(vRPCConvertParams[i].methodName,
-                                            vRPCConvertParams[i].paramName));
+    for (size_t i = 0; i < n_elem; i++) {
```

Comment:
pico-nit if you retouch (same in `src/qt/rpcconsole.cpp::L308`) 
```suggestion
    for (size_t i = 0; i < n_elem; ++i) {
```